### PR TITLE
[Feature] Support DDP Evaluation for IS and FID

### PR DESCRIPTION
**Pull request by LeoXing1996**
- [x] `slurm_eval.sh`
- [x] Update `quick_run.md`

### CODE REVIEW

File: tools/evaluation.py [hunk 1]
```diff
@@ -182,7 +213,15 @@ def main():
...
                                   args.samples_path, **args.sample_cfg)
     else:
-        raise NotImplementedError("We hasn't implemented multi gpu eval yet.")
+        find_unused_parameters = cfg.get('find_unused_parameters', False)
+        model = MMDistributedDataParallel(
+            model.cuda(),
+            device_ids=[torch.cuda.current_device()],
+            broadcast_buffers=False,
+            find_unused_parameters=find_unused_parameters)
+        single_gpu_online_evaluation(model, data_loader, metrics, logger,
```
Comment (nbei) [inline tools/evaluation.py:222]: I just find the name of this function should be revised. `single_gpu` seems redundant and not correct.

Comment (nbei) [reply tools/evaluation.py:222]: In addition, it seems that you have not considered saving images in DDP. Can we support it? Sampling and saving images in DDP may save more time.

File: tools/evaluation.py [hunk 2]
```diff
@@ -14,16 +14,30 @@
 from mmgen.models import build_model
 from mmgen.utils import get_root_logger
 
+_distributed_metrics = ['FID', 'IS']
+
 
 def parse_args():
-    parser = argparse.ArgumentParser(description='Evaluate a GAN model')
+    parser = argparse.ArgumentParser(description='Evaluate a Generation model')
```
[the remark this answers is not in the document]
Comment (LeoXing1996) [reply tools/evaluation.py:20]: `quick_run.md` is updated.

File: tools/evaluation.py [hunk 3]
```diff
@@ -81,14 +99,22 @@ def main():
...
         distributed = False
+        rank = 0
     else:
         distributed = True
         init_dist(args.launcher, **cfg.dist_params)
-
-    rank, _ = get_dist_info()
+        rank, world_size = get_dist_info()
+        cfg.gpu_ids = range(world_size)
+        assert args.online or world_size == 1, (
```
[the remark this answers is not in the document]
Comment (LeoXing1996) [reply tools/evaluation.py:115]: Offline mode now supports DDP!

File: mmgen/core/evaluation/evaluation.py [hunk 4]
```diff
@@ -297,20 +307,27 @@ def single_gpu_online_evaluation(model, data_loader, metrics, logger,
...
+        pbar = mmcv.ProgressBar(max_num_images)
     # sampling fake images and directly send them to metrics
-    for begin in range(0, max_num_images, batch_size):
-        end = min(begin + batch_size, max_num_images)
+    total_batch_size = batch_size * ws
+    for _ in range(0, max_num_images, total_batch_size):
         fakes = model(
             None,
-            num_batches=end - begin,
+            num_batches=batch_size,
```
[the remark this answers is not in the document]
Comment (plyfager) [reply mmgen/core/evaluation/evaluation.py:352]: It seems that extra images will be dropped by `feed` in `Metric`.

**Review by plyfager — APPROVED** [review]

**Pull request merged**